addons/website_slides/models/slide_channel.py

```diff
@@ -680,68 +781,129 @@ def action_redirect_to_members(self, completed=False):
-
-            to_create_values = [
-                dict(channel_id=channel.id, partner_id=partner.id)
-                for channel in to_join
-                for partner in target_partners if partner.id not in existing_map[channel.id]
-            ]
-            slide_partners_sudo = self.env['slide.channel.partner'].sudo().create(to_create_values)
-            to_join.message_subscribe(partner_ids=target_partners.ids, subtype_ids=[self.env.ref('website_slides.mt_channel_slide_published').id])
-            # Also consider re-joined members along with newly joined ones
-            return slide_partners_sudo | archived_members
-        return self.env['slide.channel.partner'].sudo()
+
+            # Unarchive existing channel partners
+            archived_channel_partners = existing_channel_partners.filtered(lambda channel_partner: not channel_partner.active)
+            unarchived_channel_partners_sudo = SlideChannelPartnerSudo
+            if member_status != 'invited' and archived_channel_partners:
+                archived_channel_partners.action_unarchive()
+                unarchived_channel_partners_sudo = archived_channel_partners.sudo()
+
+            existing_channel_partners_map = defaultdict(lambda: self.env['slide.channel.partner'])
+            for channel_partner in existing_channel_partners:
+                existing_channel_partners_map[channel_partner.channel_id] |= channel_partner
+
+            # Invited partners confirming their invitation by enrolling to the course.
+            channel_partners_joined_sudo = SlideChannelPartnerSudo
+            to_create_partners_values = []
+
+            for channel in channels_to_join:
+                channel_partners = existing_channel_partners_map[channel]
+                if member_status != 'invited':
+                    channel_partners_joined_sudo |= channel_partners.filtered(lambda cp: cp.member_status == 'invited')
+
+                for partner in target_partners.filtered(lambda p: p not in channel_partners.partner_id):
+                    to_create_partners_values.append(dict(channel_id=channel.id, partner_id=partner.id, member_status=member_status))
+
+            new_slide_channel_partners = SlideChannelPartnerSudo.create(to_create_partners_values)
+            channel_partners_joined_sudo.member_status = 'joined'
+
+            # Subscribe partners joining to the course to the chatter.
+            if member_status != 'invited':
+                for channel_partner in unarchived_channel_partners_sudo | channel_partners_joined_sudo | new_slide_channel_partners:
+                    channel_partner.channel_id.sudo().message_subscribe(
+                        partner_ids=[channel_partner.partner_id.id],
+                        subtype_ids=[self.env.ref('website_slides.mt_channel_slide_published').id]
+                    )
+            return unarchived_channel_partners_sudo | channel_partners_joined_sudo | new_slide_channel_partners
```

Comment:
A bit complicated ... if I understand well
Join: new + existing that were invited + existing that were archived
Invite: new only ?

Not sure why we add the unarchived one... they are not "new" members at enroll time if they were already enrolled. Don't we want to return only the new memberships, notably for overrides ?

Reply:
Exactly, see above for details. Also, small inventory of the 3 places this result is used:
- action_invite in the wizard -> An email is sent per scp in result
- action_grant_access (for a single partner) -> if result,  then we look for the action waiting for its feedback. ( so inactive enrolled should be unarchived and returned, right?)
- action_channel_join -> 'success' if a partner is returned -> return success. So same as last point, we should return an unarchived member enrolling themselves in a public course, for instance.

And that is it. I've removed the other method :)